test/integration/full/preload-cssom/preload-cssom.js

```diff
@@ -0,0 +1,202 @@
+/* global axe, Promise */
+describe('preload cssom integration test pass', function() {
+	'use strict';
+
+	var origAxios;
+
+	before(function(done) {
+		function start() {
+			// Stop messing with my tests Mocha!
+			document.querySelector('#mocha h1').outerHTML =
+				'<h2>preload cssom integration test</h2>';
+			// cache original axios object
+			if (axe.imports.axios) {
+				origAxios = axe.imports.axios;
+			}
+			// run axe
+			axe.run(
+				{
+					preload: true
+				},
+				function(err) {
+					assert.isNull(err);
+					done();
+				}
+			);
+		}
+		if (document.readyState !== 'complete') {
+			window.addEventListener('load', start);
+		} else {
+			start();
+		}
+	});
+
+	function createStub(shouldReject) {
+		axe.imports.axios = function stubbedAxios() {
+			return new Promise(function(resolve, reject) {
+				if (shouldReject) {
+					reject(new Error('Fake Error'));
+				}
+				resolve({
+					data: 'body{overflow:auto;}'
+				});
+			});
+		};
+	}
+
+	function restoreStub() {
+		if (origAxios) {
+			axe.imports.axios = origAxios;
+		}
+	}
+
+	function assertStylesheet(sheet, selectorText, cssText) {
+		assert.isDefined(sheet);
+		assert.property(sheet, 'cssRules');
+		assert.equal(sheet.cssRules[0].selectorText, selectorText);
+		assert.equal(sheet.cssRules[0].cssText.replace(/\s/g, ''), cssText);
+	}
+
+	function getPreload(root) {
+		var config = {
+			asset: 'cssom',
+			timeout: 10000,
+			treeRoot: axe.utils.getFlattenedTree(root ? root : document)
+		};
+		return axe.utils.preloadCssom(config);
+	}
+
+	function commonTestsForRootAndFrame(root) {
+		it('should return external stylesheet from cross-domain and verify response', function(done) {
+			getPreload(root).then(function(results) {
+				var sheets = results[0];
+				var externalSheet = sheets.filter(function(s) {
+					return s.isExternal;
+				})[0];
+				assertStylesheet(externalSheet, 'body', 'body{overflow:auto;}');
+				done();
+			});
+		});
+
+		it('should reject external stylesheets', function(done) {
+			restoreStub();
+			createStub(true);
+			var doneCalled = false;
+			getPreload(root).catch(function(error) {
+				assert.equal(error.message, 'Fake Error');
+				if (!doneCalled) {
+					done();
+				}
+				doneCalled = true;
+			});
+		});
+	}
+
+	beforeEach(function() {
+		if (window.PHANTOMJS) {
+			assert.ok('PhantomJS is a liar');
+			this.currentTest.fn = function() {
+				this.skip();
+			};
+		}
+		createStub();
+	});
+
+	afterEach(function() {
+		restoreStub();
+	});
+
+	describe('tests for current top level document', function() {
+		it('should return inline stylesheets defined using <style> tag', function(done) {
+			getPreload().then(function(results) {
```

Comment:
Do we need a `.catch()` block here?

Reply:
Well, the tests are engineered to pass/ fail. I am testing for catch later too. I expect this test to pass and not error/ throw.

Reply:
So, if I'm understanding this test correctly, if an engineer edited some part of the API this test uses and caused `getPreload()` to throw an error, we'd rather have an unhandled rejection than a "pretty" test failure?

This seems wrong and I suspect will cause someone headaches eventually.

We could just add `.catch(done)` to these Promises and prevent crashes/unhandled rejections.

Reply:
I think you have contradicted yourself here, give me a ring to discuss (easier).